scripts/tasks_retrival/gene_disease_association_task_creation.py

```diff
@@ -0,0 +1,260 @@
+        )
+    return input_file
+
+
+def get_symbols(gene_targetId_list):
+    """
+        given s list of gene id's (names Like ENSG00000006468) this method
+        uses the MyGenInfo package to retrieve the gene symbol (name like PLAC4).
+
+    Args:
+    ----
+        gene_targetId_list (list): list of gene id's (names Like ENSG00000006468)
+
+    Returns:
+    -------
+        list: List of corresponding symbols
+
+    """
+    mg = mygene.MyGeneInfo()
+    list_of_gene_metadata = mg.querymany(
+        gene_targetId_list.drop_duplicates(), species="human", fields="symbol"
+    )
+    gene_metadata_df = get_id_to_symbol_df(list_of_gene_metadata)
+    return pd.DataFrame(
+        [gene_metadata_df.loc[x, "symbol"] for x in gene_targetId_list],
+        columns=["symbol"],
+    )
+
+
+def get_id_to_symbol_df(list_of_gene_metadata):
+    """
+        The method converts a list of gene metadata into a data frame,
+        each dictionary will contain the field symbol and the gene id as the query value
+    Args:
+        list_of_gene_metadata (list): list containing gene metadata.
+
+    Returns
+    -------
+        pd.DataFrame: a data frame with the gene id as index with the symbol as value
+
+    """
+    gene_metadata_df = pd.DataFrame(list_of_gene_metadata)
+    # some target ids have multiple symbols
+    gene_metadata_df = gene_metadata_df.drop_duplicates(subset="query")
+    gene_metadata_df.index = gene_metadata_df["query"]
+    return gene_metadata_df
+
+
+def get_gda_data(input_file: str):
+    res = []
+    file_exist = True
+    part_ind = 0
+    while file_exist:
+        try:
+            link_ars = f"associationByDatasourceDirect/part-{part_ind:05}-6866be1a-be5d-40cf-bdf6-627bef1d0410-c000.snappy.parquet"
+            gda_df = pd.read_parquet(input_file + link_ars)
+            res.append(gda_df)
+            part_ind = part_ind + 1
+        except Exception as exception:
+            if not len(res):
+                raise RuntimeError(f"could not read from {input_file}") from exception
+            file_exist = False
+            break
+
+    return pd.concat(res)
+
+
+def extract_outcome_df(
+    downloaded_dataframe: pd.DataFrame, target_column_name: str = COLUMN_OF_OUTCOME
+) -> pd.DataFrame:
+    """Splice the two columns of interest into a new data frame, rename and extract HLC Class from the name."""
+    outcomes = pd.DataFrame({"Outcomes": downloaded_dataframe[target_column_name]})
+    # make sure there are no extra spaces around the values and return
+    return outcomes.map(lambda x: x.strip() if isinstance(x, str) else x)
+
+
+def report_task(df, task_dir_name):
+    print(f"Task saved to {task_dir_name}/")
+
+    MAX_COUNTS = 10
+    col_name = df.columns[0]
+    print()  # for readability of the message
+    val_counts = pd.DataFrame(df[col_name].value_counts(), columns=["count"])
+    print(val_counts.iloc[:MAX_COUNTS].to_string())
+    if len(val_counts) > MAX_COUNTS:
```

Comment:
Maybe we should just print the minimum between the number of values and max_counts.

Reply:
I find this less informative.  The print was added to account for cases where we get a large(ish) number of values, and want to report both the value counts for the top values and some indication of what remains unpainted.  It's not all that interesting in this case, as the values are actually regression values, so basically continues.  In cases where we have say 15-21 different categories, this gives a better feel of the data and a heads up if it may be wise to limit the number of categories used.

Reply:
Ok now I better understand what you did here. So if you will keep `val_counts ` as a series, you could simply do 
val_counts[max:]. I'm not sure if the plus one is needed even in this situation. 

The message should be more consistent so something like:
"Numer of rows is {} " (for all of the situations) and if max_counts is smaller then the number of categories add the names of the values. 
MAX_COUNTS should be defined in the header of the method.

Reply:
Actually, I do need both the to_str and the df in order to pretty print, otherwise I get "Name: count, dtype: int64" at the end or I do not get the "count" as column name. 

It can definitely  be done better, but I can't do that within reasonable time.  

We can, in this case, just drop the function all together and not report the counts, it's a regression task anyway.